### PR TITLE
Terra-grid now imports breakpoints from terra-breakpoints

### PR DESCRIPTION
**Pull request by AlexB98**
### Summary
Deleted the existing variables.scss file in terra-grid that was being used for breakpoints. Instead terra-grid now imports the breakpoints from a variables.scss file within terra-breakpoints. Also, changed the naming scheme of the size props.

Resolves [#2131](https://github.com/cerner/terra-core/issues/2131)

### CODE REVIEW

File: packages/terra-breakpoints/src/_variables.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+$terra-breakpoints: (
+  tiny: 0,
+  small: 544px,
+  medium: 768px,
+  large: 992px,
+  huge: 1216px,
+  enormous: 1440px
+);
```
Comment (StephenEsser) [inline packages/terra-breakpoints/src/_variables.scss]: Do we still want to keep this file in sass. Are consumers using it in their custom scss?

File: packages/terra-grid/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Changed
+* Removed variables file, breakpoints now imported from terra-breakpoints
```
Comment (StephenEsser) [inline packages/terra-grid/CHANGELOG.md]: There are breaking changes we should callout here. I don't know if there is a standard for the changelog entry yet, but previously I had listed those changes under a `### Breaking Changes` heading

Comment (AlexB98) [reply packages/terra-grid/CHANGELOG.md]: AH, I wasn't aware of that. Will do!

Comment (emilyrohrbough) [reply packages/terra-grid/CHANGELOG.md]: Here is what I link out for Changelog reference: https://keepachangelog.com/en/1.0.0/

File: packages/terra-grid/src/GridColumn.jsx [hunk 3]
```diff
@@ -27,9 +27,13 @@ const propTypes = {
   /**
    * The number of columns across all breakpoints. From 1-12(inclusively).
    */
-  col: columnRange,
+  tiny: columnRange,
+  /**
+   * The number of columns starts from enormous breakpoint. From 1-12(inclusively).
```
Comment (emilyrohrbough) [inline packages/terra-grid/src/GridColumn.jsx]: Can we update and change these prop descriptions with this breaking change? I find them to be confusing and always have to guess and check my implementation when trying to use them

File: packages/terra-grid/src/GridColumn.jsx [hunk 4]
```diff
@@ -27,9 +27,13 @@ const propTypes = {
   /**
    * The number of columns across all breakpoints. From 1-12(inclusively).
    */
-  col: columnRange,
+  tiny: columnRange,
```
Comment (emilyrohrbough) [inline packages/terra-grid/src/GridColumn.jsx]: Should tiny be required since it'll be the default?

File: packages/terra-grid/CHANGELOG.md (hunk 2 above)
Comment (StephenEsser) [reply packages/terra-grid/CHANGELOG.md]: I think the Breaking Changes section adds more clarity to the change log and helps the changes stand out. For non-breaking changes I think the Changelog reference above is a good standard.

Comment (StephenEsser) [reply packages/terra-grid/CHANGELOG.md]: There are a few other things we'll need to call out here. Including the removal of the `col` prop, the addition of the `enormous` prop,  and how the new breakpoint ranges affect this component. The old API has a different breakpoint mapping now

Comment (emilyrohrbough) [reply packages/terra-grid/CHANGELOG.md]: @StephenEsser I agree we should add a `### Breaking Change` section with the list of changes and possibly link to the upgrade guide

**Comment by emilyrohrbough** [comment]
@AlexB98 We need to add an upgrade guide for these breaking changes

File: packages/terra-grid/src/GridColumn.jsx (hunk 3 above)
Comment (AlexB98) [reply packages/terra-grid/src/GridColumn.jsx]: Ya... I wasn't truthfully sure what it meant either. Any clarification as to the significance of the numbers would be appreciated.

File: packages/terra-grid/CHANGELOG.md (hunk 2 above)
Comment (AlexB98) [reply packages/terra-grid/CHANGELOG.md]: Sure thing, will add a list!

**Comment by AlexB98** [comment]
@emilyrohrbough What does this entail?

**Comment by AlexB98** [comment]
> @AlexB98 We need to add an upgrade guide for these breaking changes

@emilyrohrbough What does this entail?

**Comment by emilyrohrbough** [comment]
I would expect a new section at the top of the existing upgrade guide that explains to consumers what changed and what code changes are required for consumer terra-grid v6.

https://engineering.cerner.com/terra-core/#/components/terra-grid/grid/upgrade-guide

File: packages/terra-grid/src/GridColumn.jsx (hunk 3 above)
Comment (emilyrohrbough) [reply packages/terra-grid/src/GridColumn.jsx]: As you resize thew width of your browser, the grid should re-adjust it's layout when a new breakpoint is reach. This layout will reflect column values provided to the props. I'd update an example and test it out a bit to find a description that makes the best sense to you

Comment (bjankord) [reply packages/terra-grid/src/GridColumn.jsx]: I'd recommend the following for the prop description.
```
Defines the number of columns this container will span in the 12 column grid. It is applied at the `tiny` breakpoint and wider screens.
```

**Comment by AlexB98** [comment]
> Do we have wdio tests for each breakpoint size? I noticed the medium screenshots were updated

Updated the screenshots I saw that used the now nonexistent 'col' prop, and added the 'enormous' prop.

File: packages/terra-grid/src/GridColumn.jsx (hunk 4 above)
Comment (bjankord) [reply packages/terra-grid/src/GridColumn.jsx]: This should stay as an optional prop.

**Comment by AlexB98** [comment]
@StephenEsser @emilyrohrbough Any more changes/fixes you would like to see?

**Comment by StephenEsser** [comment]
@AlexB98 

This PR was brought up and discussed in our war-room. It sounded like Brett would sync up with you to discuss any potential changes. 

cc: @bjankord

**Comment by bjankord** [comment]
@StephenEsser I've updated this branch with additions we discussed in our meeting on Friday here: https://github.com/cerner/terra-core/pull/2373/commits/a6dc28763867f417d20553f2f40c6dd761c6dd58

@emilyrohrbough I've created test pages to cover the various column width props here: https://github.com/cerner/terra-core/pull/2373/commits/50e05c062b03d02fb6968505fe8a7586d179994c
@AlexB98 We'll want to get wdio tests in place to capture test evidence of these new pages.

**Pull request merged**